Latest bleak changed API, tests are failing

```
~/src/idasen-ha$ python3 -m pytest -x
ImportError while loading conftest '/home/edward/src/2025/vendor/idasen-ha/tests/conftest.py'.
tests/__init__.py:6: in <module>
    FAKE_BLE_DEVICE = BLEDevice("AA:BB:CC:DD:EE:FF", None, {"path": ""}, 0)
E   TypeError: BLEDevice.__init__() takes 4 positional arguments but 5 were given
~/src/idasen-ha$
```

I tried removing the last parameter from the call to BLEDevice() in the tests.

```
~/src/idasen-ha$ git diff
diff --git a/tests/__init__.py b/tests/__init__.py
index f825d9e..cfa6024 100644
--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -3,7 +3,7 @@
 from bleak.backends.device import BLEDevice
 from idasen import IdasenDesk
 
-FAKE_BLE_DEVICE = BLEDevice("AA:BB:CC:DD:EE:FF", None, {"path": ""}, 0)
+FAKE_BLE_DEVICE = BLEDevice("AA:BB:CC:DD:EE:FF", None, {"path": ""})
 
 
 def height_percent_to_meters(percent: float):
~/src/idasen-ha$
```

I now get this:

```
~/src/idasen-ha$ python3 -m pytest -x
/usr/lib/python3/dist-packages/pytest_asyncio/plugin.py:207: PytestDeprecationWarning: The configuration option "asyncio_default_fixture_loop_scope" is unset.
The event loop scope for asynchronous fixtures will default to the fixture caching scope. Future versions of pytest-asyncio will default the loop scope for asynchronous fixtures to function scope. Set the default fixture loop scope explicitly in order to avoid unexpected behavior in the future. Valid fixture loop scopes are: "function", "class", "module", "package", "session"

  warnings.warn(PytestDeprecationWarning(_DEFAULT_FIXTURE_LOOP_SCOPE_UNSET))
============================================================================================================================= test session starts ==============================================================================================================================
platform linux -- Python 3.13.5, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/edward/src/2025/vendor/idasen-ha
configfile: pytest.ini
plugins: cov-5.0.0, requests_mock-1.12.1, anyio-4.8.0, aresponses-3.0.0, asyncio-0.25.1, mock-3.14.0, django-4.5.2, time-machine-2.16.0, timeout-2.3.1, pytest_freezer-0.4.9, xvfb-3.0.0, unordered-0.7.0, xdist-3.6.1, socket-0.7.0, tornasync-0.6.0.post2, hypothesis-6.130.5, typeguard-4.4.2, syrupy-4.9.1, Faker-33.3.1, kgb-7.2, respx-0.22.0
asyncio: mode=Mode.AUTO, asyncio_default_fixture_loop_scope=None
collected 26 items                                                                                                                                                                                                                                                             

tests/test_connection_management.py F

=================================================================================================================================== FAILURES ===================================================================================================================================
___________________________________________________________________________________________________________________________ test_connect_disconnect ____________________________________________________________________________________________________________________________

mock_idasen_desk = <async_generator object mock_idasen_desk at 0x7f791686f9a0>

    async def test_connect_disconnect(mock_idasen_desk: MagicMock):
        """Test connect and disconnect."""
        update_callback = Mock()
        desk = Desk(update_callback, False)
    
        await desk.connect(FAKE_BLE_DEVICE)
>       assert desk.is_connected
E       assert False
E        +  where False = <idasen_ha.Desk object at 0x7f7916768d70>.is_connected

tests/test_connection_management.py:23: AssertionError
------------------------------------------------------------------------------------------------------------------------------ Captured log call -------------------------------------------------------------------------------------------------------------------------------
WARNING  idasen:__init__.py:166 [AA:BB:CC:DD:EE:FF] Failed to connect, retrying (1/3)...
WARNING  idasen:__init__.py:166 [AA:BB:CC:DD:EE:FF] Failed to connect, retrying (2/3)...
WARNING  idasen:__init__.py:166 [AA:BB:CC:DD:EE:FF] Failed to connect, retrying (3/3)...
CRITICAL idasen:__init__.py:161 [AA:BB:CC:DD:EE:FF] Connection failed
ERROR    idasen_ha.connection_manager:connection_manager.py:70 Connect failed
Traceback (most recent call last):
  File "/home/edward/src/2025/vendor/idasen-ha/idasen_ha/connection_manager.py", line 68, in _connect
    await self._idasen_desk.connect()
  File "/usr/lib/python3/dist-packages/idasen/__init__.py", line 156, in connect
    await self._client.connect()
  File "/usr/lib/python3/dist-packages/bleak/__init__.py", line 566, in connect
    await self._backend.connect(self._pair_before_connect, **kwargs)
  File "/usr/lib/python3/dist-packages/bleak/backends/bluezdbus/client.py", line 190, in connect
    watcher = manager.add_device_watcher(
        self._device_path, on_connected_changed, on_value_changed
    )
  File "/usr/lib/python3/dist-packages/bleak/backends/bluezdbus/manager.py", line 624, in add_device_watcher
    self._check_device(device_path)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/bleak/backends/bluezdbus/manager.py", line 206, in _check_device
    raise BleakError(f"device '{device_path.split('/')[-1]}' not found")
bleak.exc.BleakError: device '' not found
=========================================================================================================================== short test summary info ============================================================================================================================
FAILED tests/test_connection_management.py::test_connect_disconnect - assert False
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================================================================================== 1 failed in 2.03s ===============================================================================================================================
~/src/idasen-ha$
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,7 +32,7 @@ exclude = ["tests"] # tests.*
 namespaces = false
 
 [tool.ruff]
-select = [
+lint.select = [
     "B002", # Python does not support the unary prefix increment
     "B007", # Loop control variable {name} not used within loop body
     "B014", # Exception handler with duplicate exception
@@ -52,7 +52,6 @@ select = [
     "N804", # First argument of a class method should be named cls
     "N805", # First argument of a method should be named self
     "N815", # Variable {name} in class scope should not be mixedCase
-    "PGH001", # No builtin eval() allowed
     "PGH004",  # Use specific rule codes when using noqa
     "PLC0414", # Useless import alias. Import alias does not rename original package.
     "PLC", # pylint
@@ -73,7 +72,6 @@ select = [
     "S317",  # suspicious-xml-sax-usage
     "S318",  # suspicious-xml-mini-dom-usage
     "S319",  # suspicious-xml-pull-dom-usage
-    "S320",  # suspicious-xmle-tree-usage
     "S601",  # paramiko-call
     "S602",  # subprocess-popen-with-shell-equals-true
     "S604",  # call-with-shell-equals-true
@@ -91,13 +89,13 @@ select = [
     "T20",  # flake8-print
     "TID251", # Banned imports
     "TRY004", # Prefer TypeError exception for invalid type
-    "TRY200", # Use raise from to specify exception cause
-    "TRY302", # Remove exception handler; error is immediately re-raised
+    "S307", # Use raise from to specify exception cause
+    "TRY203", # Remove exception handler; error is immediately re-raised
     "UP",  # pyupgrade
     "W",  # pycodestyle
 ]
 
-ignore = [
+lint.ignore = [
     "D202",  # No blank lines allowed after function docstring
     "D203",  # 1 blank line required before class docstring
     "D213",  # Multi-line docstring summary should start at the second line
@@ -121,10 +119,10 @@ ignore = [
 
 ]
 
-[tool.ruff.isort]
+[tool.ruff.lint.isort]
 force-sort-within-sections = true
 combine-as-imports = true
 split-on-trailing-comma = false
 
-[tool.ruff.mccabe]
+[tool.ruff.lint.mccabe]
 max-complexity = 25

--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -3,7 +3,7 @@
 from bleak.backends.device import BLEDevice
 from idasen import IdasenDesk
 
-FAKE_BLE_DEVICE = BLEDevice("AA:BB:CC:DD:EE:FF", None, {"path": ""}, 0)
+FAKE_BLE_DEVICE = BLEDevice("AA:BB:CC:DD:EE:FF", None, {"path": ""})
 
 
 def height_percent_to_meters(percent: float):

--- a/tests/test_connection_management.py
+++ b/tests/test_connection_management.py
@@ -64,7 +64,7 @@ async def test_double_connect_call_with_different_bledevice():
         async def connect_side_effect():
             # call the seccond `connect` while the first is ongoing
             mock_idasen_desk.connect.side_effect = MagicMock()
-            new_ble_device = BLEDevice("AA:BB:CC:DD:EE:AA", None, None, 0)
+            new_ble_device = BLEDevice("AA:BB:CC:DD:EE:AA", None, None)
             await desk.connect(new_ble_device)
 
         mock_idasen_desk.is_connected = False

